Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/BehaviorData.xml

```diff
@@ -4688,6 +4688,24 @@
             <Chance value="1"/>
         </DamageResponse>-->
     </CBehaviorBuff>
+    <CBehaviorBuff id="AP_MothershipTaldarimMovementNerfOutsidePower">
+        <Alignment value="Negative"/>
+        <BehaviorFlags index="Permanent" value="1"/>
+        <InfoIcon value="Assets\Textures\btn-ability-stetmann-stetzonegenerator-energy.dds"/>
+        <EditorCategories value="AbilityorEffectType:Units"/>
+        <DisableValidatorArray value="AP_NotHaveMothershipTaldarimIntegratedPower"/>
+        <DisableValidatorArray value="AP_CasterLocationNotHasPower"/>
+        <Modification MoveSpeedMultiplier="0.2"/>
+    </CBehaviorBuff>
+    <CBehaviorPowerSource id="AP_DisplayPowerFieldMothership">
+        <InfoFlags index="Hidden" value="1"/>
+        <InfoIcon value="Assets\Textures\btn-building-protoss-pylon.dds"/>
+        <EditorCategories value="Race:Protoss,AbilityorEffectType:Structures"/>
+        <DisableValidatorArray value="AP_NotHaveMothershipTaldarimIntegratedPower"/>

```

Comment:
Editor Warning:
`Warning: XML: Z:\hdd\home\venca\software\battle-net\drive_c\Program Files (x86)\StarCraft II\Mods\ArchipelagoPlayer.SC2Mod\Base.SC2Data\GameData\BehaviorData.xml(4704,9) : Unable to find field DisableValidatorArray  Source: Mods/ArchipelagoPlayer.SC2Mod  Entry: AP_DisplayPowerFieldMothership  Scope: CBehaviorPowerSource`

Reply:
I believe this is the root of your bug as the behavior isn't a `CBehaviorBuff`. You can, however remove the behavior with a trigger (There's a trigger that removes Ursadon's `CBehaviorLeash` on being derpfestor'd)
You want to remove the behavior on following events: An upgrade is being given or a unit is trained

Reply:
Alternatively, you could have a second behavior that's acting as a controller and when exiting, do the remove behavior effect that removes the power source

Reply:
BTW found that Power source behavior can use requirements:
```xml
    <CBehaviorPowerSource id="AP_WarpPrismPowerSource">
        <InfoFlags index="Hidden" value="1"/>
        <EditorCategories value="Race:Protoss,AbilityorEffectType:Structures"/>
        <Radius value="3.75"/>
        <PowerLink value="PowerSource"/>
        <Requirements value="AP_NotHaveWarpPrismWarConfiguration"/>
    </CBehaviorPowerSource>
 ```

Reply:
So I've tried
* Requirements
* DisableValidator
* RemoveValidator
* Removing straight from the upgrade's EffectArray
* Having another behaviour remove it with BehaviorLinkDisableArray

Some of those removed the behaviour icon, but none removed the power field visuals. As far as I can tell, it's the presence of a link to this behaviour type at all that works.

Checking the triggers, `UnitBehaviorRemove` seems to target a particular unit rather than a unit type. I'll try tinkering with a behaviour chain before I go the trigger route, as triggers likely means a persistent trigger that enables / disables behaviours continuously.

Reply:
I think I figured out how to approach this. Buried all the way in core is this:
```xml
    <CActorPower default="1">
        <PowerSource value="##id##Source"/>
    </CActorPower>
```
Which seems to be the root of the display woes. I should be able to trace this through actordata and deal with the visuals from there.

Reply:
Opened things in the editor to check, all error messages related to this PR should be cleaned up
![image](https://github.com/user-attachments/assets/5e9b89ab-aaa8-45b8-b93e-b0a6ff5bb26a)